File: lab/javascript/matematica_e_operadores/contagem_regressiva_recursiva.js
function contagem_recursiva(n) {
  if (n === 1) {
    console.log(n);
    return n;
  }
  process.stdout.write(n + " ");
  return n - contagem_recursiva(n - 1);
}

function contagem_recursiva2(n) {
  if (n === 0) {
    return;
  }
  process.stdout.write(n + " ");
  contagem_recursiva(n - 1);
}

// Testes
contagem_recursiva(5);
contagem_recursiva2(5);

File: lab/c/matematica_e_operadores/contagem_regressiva_recursiva.c
#include <stdio.h>

int contagem_recursiva(int n) {
    if (n == 1) {
        printf("%d\n", n);
        return n;
    }
    printf("%d ", n);
    return n - contagem_recursiva(n - 1);
}

void contagem_recursiva2(int n) {
    if (n == 0)
        return;
    printf("%d ", n);
    contagem_recursiva(n - 1);
}

int main(void) {
    contagem_recursiva(5);
    contagem_recursiva2(5);
    return 0;
}
